bundle/jsky.app.ot/src/main/scala/jsky/app/ot/gemini/editor/EphemerisUpdater.scala

```diff
@@ -105,16 +107,12 @@ object EphemerisUpdater {
     }
 
   /**
-   * Refresh the ephimerides for all nonsidereal targets in `obs`, on another thread, showing status
-   * in the root pane associated with the given `Component`.
+   * Refresh the ephemerides for all nonsidereal targets in `obs`, showing status in the root pane
+   * associated with the given `Component`, logging failures.
    */
-  def unsafeRefreshEphemerides(obsN: ISPObservation, c: Component): Unit = {
+  def refreshEphemerides(obsN: ISPObservation, c: Component): IO[Unit] = {
     val ui = UI(c)
-    val action = (refreshEphemerides(obsN, ui) ensuring ui.hide).withResultLogging(Log)
-    Task(action.run.unsafePerformIO).unsafePerformAsync {
-      case -\/(t) => Swing.onEDT(DialogUtil.error(c, t))
-      case \/-(_) => () // done!
-    }
+    (refreshEphemerides(obsN, ui).withResultLogging(Log) ensuring ui.hide).run.void
```

Comment:
I've seen ensuring on code completion many times. What does it do? is it like a `finally`?